lib/terraform/runner/response_async.rb

```diff
@@ -0,0 +1,67 @@
+module Terraform
+  class Runner
+    class ResponseAsync
+      include Vmdb::Logging
+
+      # Response object designed for holding full response from terraform-runner stack job
+      #
+      # @param stack_id [String] terraform-runner stack job - stack_id
+      def initialize(stack_id)
+        @stack_id = stack_id
+      end
+
+      # @return [Boolean] true if the terraform stack job is still running, false when it's finished
+      def running?
+        return false if @response && completed?(@response.status)
+
+        # re-fetch response
+        refresh_response
+
+        !completed?(@response.status)
+      end
+
+      # Stops the running Terraform job
+      def stop
+        raise "No job running to stop" if !running?
+
+        Terraform::Runner.stop_async(@stack_id)
+      end
+
+      # Re-Fetch async job's response
+      def refresh_response
+        @response = Terraform::Runner.fetch_result_by_stack_id(@stack_id)
+
+        @response
```

Comment:
Really minor but the "return" of an assignment is the value anyway so you don't need this
```suggestion
```